MediaBrowser.Common/Extensions/HttpClientExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿using System.Net.Http;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+using MediaBrowser.Common.Json;
+
+namespace MediaBrowser.Common.Extensions
+{
+    /// <summary>
+    /// Http Client Extensions.
+    /// </summary>
+    public static class HttpClientExtensions
+    {
+        private static readonly JsonSerializerOptions _defaultJsonOptions = JsonDefaults.GetOptions();
+
+        /// <summary>
+        /// Read json from uri as T.
+        /// </summary>
+        /// <param name="client">The http client.</param>
+        /// <param name="requestUri">The request uri.</param>
+        /// <param name="jsonSerializerOptions">The json serializer options.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <typeparam name="T">Type to deserialize json as.</typeparam>
+        /// <returns>Deserialized json.</returns>
+        public static async Task<T> ReadAsAsync<T>(
+            this HttpClient client,
+            string requestUri,
+            JsonSerializerOptions jsonSerializerOptions = null,
+            CancellationToken cancellationToken = default)
+        {
+            using var response = await client.GetAsync(requestUri, HttpCompletionOption.ResponseHeadersRead, cancellationToken).ConfigureAwait(false);
+            return await response.ReadAsAsync<T>(jsonSerializerOptions, cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Read json from http response message as T.
+        /// </summary>
+        /// <param name="responseMessage">The response message.</param>
+        /// <param name="jsonSerializerOptions">The json serializer options.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <typeparam name="T">Type to deserialize json as.</typeparam>
+        /// <returns>Deserialized json.</returns>
+        public static async Task<T> ReadAsAsync<T>(
+            this HttpResponseMessage responseMessage,
+            JsonSerializerOptions jsonSerializerOptions = null,
+            CancellationToken cancellationToken = default)
+        {
+            responseMessage.EnsureSuccessStatusCode();
```

Comment:
I was a bit busy yesterday, so I didn't finish my review. I have a few comments:

1. This is not a good idea. Most APIs will return a 4xx or 5xx code with an error message, but that message is gone if you use this as it just throws a `HttpRequestException`.
Handling errors should be done with `HttpResponseMessage.IsSuccessStatusCode`, which leads into a second point about the first extension method.
2. The first extension method hides the response from the caller, which is needed for error handling, I therefore see very limited use for it.